DMCompiler/DM/DMProc.cs

```diff
@@ -826,15 +826,19 @@ public void Pop()
             WriteByte((byte)reference.RefType);
 
             switch (reference.RefType) {
-                case DMReference.Type.Argument: WriteByte(reference.ArgumentId); break;
-                case DMReference.Type.Local: WriteByte(reference.LocalId); break;
-                case DMReference.Type.Global: WriteInt(reference.GlobalId); break;
-                case DMReference.Type.Field: WriteString(reference.FieldName); ShrinkStack(affectStack ? 1 : 0); break;
-                case DMReference.Type.SrcField: WriteString(reference.FieldName); break;
-                case DMReference.Type.Proc: WriteString(reference.ProcName); ShrinkStack(affectStack ? 1 : 0); break;
-                case DMReference.Type.GlobalProc: WriteString(reference.ProcName); break;
-                case DMReference.Type.SrcProc: WriteString(reference.ProcName); break;
+                case DMReference.Type.Argument:
+                case DMReference.Type.Local:
```

Comment:
Arguments and Locals have a max of 255 and can be written as bytes rather than integers. Writing them as integers appears to make the output of stripped /tg/ ~156 KB larger and TestGame ~3 KB larger.

It would be a good idea to rename `DMReference.Int` to `DMReference.Index` to more accurately reflect what it is.
```suggestion
                case DMReference.Type.Argument:
                case DMReference.Type.Local: WriteByte((byte)reference.Int); break;
                
```